### PR TITLE
Fix pruning with LruCaheLowObject

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Improve type conversions
- Use DisableWAL when pruning old entries (will never be read again)
- Increase pruning ConcurrentDictionary key set
- Batch deletes into smaller chunks (256 rather than 250k)

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 1]
```diff
@@ -787,33 +790,48 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, in Val
                 return true;
             }
 
-            using INodeStorage.WriteBatch writeBatch = _nodeStorage.StartWriteBatch();
+            ActionBlock<INodeStorage.WriteBatch> actionBlock =
+                new ActionBlock<INodeStorage.WriteBatch>((batch) => batch.Dispose());
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:794]: make delegate explicitly static?

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 2]
```diff
@@ -787,33 +790,48 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, in Val
...
+                _logger.Error($"Failed to remove past keys. {ex}");
+            }
+            finally
+            {
+                writeBatch.Dispose();
             }
 
             actionBlock.Complete();
             actionBlock.Completion.Wait();
+            _nodeStorage.Compact();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:834]: shouldn't those 3 lines be in finally?

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 3]
```diff
@@ -787,33 +790,48 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, in Val
...
             }
-
-            ActionBlock<KeyValuePair<HashAndTinyPath, Hash256>> actionBlock =
-                new ActionBlock<KeyValuePair<HashAndTinyPath, Hash256>>(DoAct);
-
-            foreach (KeyValuePair<HashAndTinyPath, Hash256> keyValuePair in persistedHashes)
+            catch (Exception ex)
             {
-                actionBlock.Post(keyValuePair);
+                _logger.Error($"Failed to remove past keys. {ex}");
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:825]: IsError

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 4]
```diff
@@ -787,33 +790,48 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, in Val
...
+                        if (CanRemove(key.addr, key.path, fullPath, prevHash, keyValuePair.Value))
+                        {
+                            Metrics.RemovedNodeCount++;
+                            writeBatch.Set(key.addr, fullPath, prevHash, default, WriteFlags.DisableWAL);
+                            round++;
+                        }
+                    }
+
+                    // Batches of 1000
+                    if (round > 1000)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:815]: did you experiment with other batch sizes?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs (hunk 3 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs]: Added

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs]: Moved

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs]: Done

File: src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs [hunk 5]
```diff
@@ -201,7 +201,7 @@ public class DbConfig : IDbConfig
     public int? StateDbBloomFilterBitsPerKey { get; set; } = 15;
     public int? StateDbUseRibbonFilterStartingFromLevel { get; set; } = 2;
     public double? StateDbDataBlockIndexUtilRatio { get; set; } = 0.5;
-    public bool StateDbEnableFileWarmer { get; set; } = false;
+    public bool StateDbEnableFileWarmer { get; set; } = true;
```
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs]: Yep

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs (hunk 4 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:817]: Testing with 256 atm

**Pull request merged**